web/views.py

```
@@ -7018,3 +7037,102 @@ def users_list(request: HttpRequest) -> HttpResponse:
+            {"success": True, "message": f"Custom badge '{badge.name}' awarded successfully", "badge_id": badge.id}
+        )
+    except (User.DoesNotExist, Course.DoesNotExist) as e:
+        return JsonResponse({"success": False, "message": str(e)}, status=404)
+    except Exception as e:
+        logger.exception(f"Error awarding custom badge: {e!s}")
+        return JsonResponse({"success": False, "message": "An unexpected error occurred"}, status=500)
+
+
+@login_required
+@require_GET
+def all_badges(request):
```

Ruff (0.8.2)

Missing return type annotation for public function `all_badges`

(ANN201)